include/my_rdtsc.h

```diff
@@ -97,7 +97,7 @@ C_MODE_START
 /**
   A cycle timer.
 
-  On clang we use __builtin_readcyclecounter(), except for AARCH64.
+  On clang we use __builtin_readcyclecounter(), except for AARCH64 and RISC-V.
```

Comment:
It would be nice to see a justification for these exceptions, such as links to clang bug reports. I found that the RISC-V implementation was added in https://reviews.llvm.org/D64125 but I found no bug report about this. Can you please file one?

For ARMv8 there already is a bug report https://github.com/llvm/llvm-project/issues/43652 where it is suggested that the instruction could work depending on the operating system kernel configuration.

These optionally privileged instructions could be usable if on the affected ISA we made `my_timer_cycles()` a function pointer. An initialization function could try to run the instruction under a SIGILL wrapper.

Reply:
Quite right, was thinking the same. Clang isn't just for kernels and root processes. I did compiler export the latest version and the behaviour is still there.